REB_firmware/REB_firmware_update/v0/producer_REB_firmware_update.py

```
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+from ccsTools import ccsProducer
+from java.lang import Exception
+import Tkinter
+import glob
+import os
+import subprocess
+
+CCS.setThrowExceptions(True);
+
+#Copy firmware_load_scripts files
+instdir = os.getenv('LCATR_INSTALL_AREA').replace('share/','')
+os.system('cp -vp %s/REB_v5/firmware_load_scripts/bin/et_prog_flash.sh .', instdir)
+os.system('cp -vp %s/REB_v5/firmware_load_scripts/bin/flash_load.tcl .', instdir)
+
+#Safety button
+apptxt = "Click here when ready to start firmware update of the REB"
+print apptxt
+top = Tkinter.Tk()
+A = Tkinter.Button(top, text = apptxt, command = top.destroy, bg = "yellow", font = ("Helvetica",24))
+A.pack()
+top.title('Are you ready to start the firmware update?')
+top.mainloop()
+
+#Get list of files in firmware directory
+list_of_firmware_files = glob.glob(os.path.join(instdir,'/REB_v5/targets/REB_v5_top/images/', '*.mcs'))
+
+#Find most recent file
+recent_file = max(list_of_firmware_files, key=os.path.getctime)
+    
+#Call Stefano's REB5 firmware upgrade program
+try:
+    subprocess.call(['source et_prog_flash.sh ', '/path/to/directory/' + recent_file])
```

I would change this to `subprocess.check_call` and remove the `try/except`. As it is, the line will not produce an exception in case the shell command fails.  One would need to check the return code explicitly; using `check_call` does that for you.  In general, if an exception is raised, it's better to let it fail unhandled so that the specific exception is reported rather than printing a message that may or may not be relevant.  If there are algorithmic reasons to handle specific exceptions, then using a `try/except` block like this is fine.